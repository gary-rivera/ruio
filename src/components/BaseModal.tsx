import React, { ReactNode } from 'react'
// import styles from './BaseModal.module.css'

type BaseModalProps = {
  isOpen: boolean
  onClose: () => void
  title?: string
  children: ReactNode
  footer?: ReactNode
}

function BaseModal({ isOpen, onClose, title, children, footer }: BaseModalProps) {
  if (!isOpen) return null

  return (
    <div
      style={{ width: '75px', height: '40px', backgroundColor: '#eaf8ef', border: '1px solid #eaf8ef' }}
    >
      <header>{title}</header>
      <section>{children}</section>
      {footer && <div>{footer}</div>}
    </div>
  )
}

export default BaseModal
