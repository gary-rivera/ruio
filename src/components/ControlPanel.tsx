import React from 'react'
import { useRuioContext } from '@context/RuioContextProvider'

const ControlPanel: React.FC = () => {
  const { depth, setDepth, ruioEnabled, setRuioEnabled, toggleElementSelectionMode } = useRuioContext()

  const handleDepthChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    setDepth(parseInt(event.target.value, 10))
  }

  return (
    <div
      data-testid="ruio-control-panel"
      className="ruio-exclude"
      style={{
        position: 'fixed',
        bottom: '80px',
        right: '20px',
        zIndex: 9999,
        backgroundColor: 'white',
        padding: '10px',
        border: '1px solid black',
      }}
    >
      <label className="ruio-exclude">
        Depth:
        <input
          className="ruio-exclude"
          type="number"
          value={depth}
          onChange={handleDepthChange}
          min="1"
          max="10"
        />
      </label>

      <div className="ruio-exclude">
        <button className="ruio-exclude" onClick={toggleElementSelectionMode} disabled={!ruioEnabled}>
          Select Element
        </button>
        {/* TODO: remove this button, its redundant with the icon */}
        <button onClick={() => setRuioEnabled(!ruioEnabled)}>
          {ruioEnabled ? 'Disable Borders' : 'Enable Borders'}
        </button>
      </div>
    </div>
  )
}

export default ControlPanel
