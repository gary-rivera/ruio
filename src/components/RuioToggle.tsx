import React from 'react'
import '@styles/RuioToggle.css'

import { useRuioContext } from '../context/RuioContextProvider'

function RuioToggle(props: React.SVGProps<SVGSVGElement>) {
  const { ruioEnabled, setRuioEnabled } = useRuioContext()

  return (
    <div
      className="ruio-exclude ruio-toggle-container"
      onClick={() => {
        setRuioEnabled(!ruioEnabled)
      }}
    >
      <svg
        data-testid="ruio-toggle-icon"
        className="ruio-exclude ruio-logo"
        viewBox="0 0 313 234"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect width="313" height="234" rx="50" fill="#264653" />

        <svg
          x="35"
          y="29"
          width="243"
          height="176"
          viewBox="0 0 243 176"
          fill="none"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            fillRule="evenodd"
            clipRule="evenodd"
            d="M202.425 19.5556H40.5748C29.4014 19.5556 20.3435 28.3109 20.3435 39.1111V136.889C20.3435 147.689 29.4014 156.444 40.5748 156.444H72.5403V56.8272L77.3229 53.959C92.2899 44.9831 130.874 32.3035 166.984 53.959L167.405 54.2116L167.799 54.5029C173.92 59.0309 181.874 67.5177 184.303 78.5538C185.567 84.2978 185.293 90.6403 182.525 97.053C179.836 103.28 175.043 109.009 168.172 114.168C165.072 116.83 160.261 119.017 154.543 120.468C152.649 120.949 150.581 121.369 148.334 121.704L177.36 156.444H202.425C213.599 156.444 222.657 147.689 222.657 136.889V39.1111C222.657 28.3109 213.599 19.5556 202.425 19.5556ZM110.468 101.581L153.159 156.444H153.172L166.414 176H202.425C224.772 176 242.888 158.489 242.888 136.889V39.1111C242.888 17.5106 224.772 0 202.425 0H40.5748C18.2279 0 0.112183 17.5106 0.112183 39.1111V136.889C0.112183 158.489 18.2279 176 40.5748 176H72.5403H92.7716V156.444V68.0911C106.309 61.9255 132.118 56.3262 155.874 70.3124C159.643 73.2142 163.475 77.8988 164.514 82.6223C165.001 84.8357 164.9 87.0982 163.854 89.522C162.776 92.019 160.439 95.2646 155.503 98.9162L155.034 99.2631L154.614 99.6571C154.721 99.5567 154.718 99.5585 154.582 99.6351C154.184 99.8603 152.644 100.732 149.405 101.554C145.373 102.577 139.231 103.378 130.705 102.845L110.468 101.581Z"
            fill="#EAF8EF"
          />
        </svg>
      </svg>
    </div>
  )
}

export default RuioToggle
