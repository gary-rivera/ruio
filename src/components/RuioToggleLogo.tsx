import React from 'react'
import '@styles/RuioToggleLogo.css'
import RuioLogoIcon from '@assets/RuioLogoIcon'

import { useRuioContext } from '../context/RuioContextProvider'

// NOTE: semi redundant, but this is a more specific component for the logo and its management of the toggle state whereas RuioLogoIcon is just the icon itself
function RuioToggleLogo() {
  const { ruioEnabled, setRuioEnabled } = useRuioContext()

  return (
    <div
      className="ruio-exclude ruio-toggle-logo"
      onClick={() => {
        setRuioEnabled(!ruioEnabled)
        console.log('RuioToggleLogo clicked', ruioEnabled)
      }}
    >
      <RuioLogoIcon />
    </div>
  )
}

export default RuioToggleLogo
