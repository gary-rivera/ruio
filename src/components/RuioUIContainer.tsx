import React, { forwardRef } from 'react'
import RuioToggleController from '../controllers/RuioToggleController'

import divStyles from '@styles/Div.module.css'
import SettingsIcon from '@assets/SettingsIcon'
import ToggleElementSelectionModeIcon from '@assets/ToggleElementSelectionModeIcon'

interface RuioUIContainerProps {
  // NOTE: define as needed
}

function RuioUIContainer(props: RuioUIContainerProps, ref: React.Ref<HTMLDivElement>) {
  // TODO: delete once depth controlled by settings modal is built out
  // basically just ensures that the borders are applied to the root element on initial load to the starting point
  // useEffect(() => {
  //   const rootElement = document.querySelector('#root')

  //   if (rootElement) applyOutlineUI(rootElement as HTMLElement, depth, ruioEnabled)
  // }, [ruioEnabled, depth])

  return (
    <div
      ref={ref}
      data-testid="ruio-ui-container"
      className={`
        ruio exlcude
        ${divStyles['ruio-ui-container']}
      `}
    >
      {/* all ruio components are lazy-loaded, css controls respective render effects */}
      <RuioToggleController />
      {ruioEnabled && (
        <div className="ruio-exclude ruio-icons-container">
          <SettingsIcon />
          <ToggleElementSelectionModeIcon />
        </div>
      )}
    </div>
  )
}

export default forwardRef<HTMLDivElement, RuioUIContainerProps>(RuioUIContainer)
