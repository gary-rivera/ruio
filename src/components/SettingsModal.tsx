import { ReactNode } from 'react'
import dividerStyles from '../styles/HorizontalDivider.module.css'

type SettingsModalProps = {
  isOpen: boolean
  onClose: () => void
  title?: string
  footer?: ReactNode
  position: { right: number; bottom: number }
}

function SettingsModal({ isOpen, onClose, position, title, footer }: SettingsModalProps) {
  if (!isOpen) return null
  console.log('[SettingsModal] received coordinates: ', {
    right: position.right,
    bottom: position.bottom,
    offsetRight: position.right - 20,
    offsetBottom: position.bottom - 20,
  })
  return (
    <div
      className="ruio-exclude"
      style={{
        display: 'flex',
        backgroundColor: 'black',
        color: 'white',
        maxWidth: '300px',
        width: '300px',
        maxHeight: '200px',
        height: '200px',
        padding: '20px',
        borderRadius: '10px',
        position: 'absolute',
        right: '34px',
        bottom: '34px',
        flexDirection: 'column',
      }}
    >
      <header style={{ display: 'inline-flex', justifyContent: 'space-between' }}>
        <h2>Settings</h2>
        <button style={{ all: 'unset' }} onClick={onClose}>
          X
        </button>
      </header>
      <hr style={{ width: 'inherit' }} />
      <div style={{}}></div>
      <section style={{ display: 'flex', flexDirection: 'column' }}>
        <div style={{ display: 'inline-flex', alignItems: 'center', justifyContent: 'space-between' }}>
          <h3 style={{ margin: '0px' }}> depth</h3>
          <span> depth controls</span>
        </div>
        <div style={{ display: 'inline-flex', alignItems: 'center', justifyContent: 'space-between' }}>
          <h3 style={{ margin: '0px' }}> theme</h3>
          <span> theme controls</span>
        </div>
        <div style={{ display: 'inline-flex', alignItems: 'center', justifyContent: 'space-between' }}>
          <h3 style={{ margin: '0px' }}> outline</h3>
          <span> outline controls</span>
        </div>
      </section>
      <footer style={{ alignSelf: 'end' }}></footer>
    </div>
  )
}

export default SettingsModal
