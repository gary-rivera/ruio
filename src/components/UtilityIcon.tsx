import React, { useEffect } from 'react'
import { useRuioContext } from '../context/RuioContextProvider'
import { applyBorders } from '../utils/applyBorders'

type UtilityIconProps = {
  selector?: string // TODO: stricter typing of either element tag name or selector id/class
}

const UtilityIcon: React.FC<UtilityIconProps> = ({ selector = '#root' }) => {
  const { bordersEnabled, setBordersEnabled, depth } = useRuioContext()

  useEffect(() => {
    const rootElement = document.querySelector(selector)

    if (rootElement) applyBorders(rootElement as HTMLElement, depth, bordersEnabled)
  }, [bordersEnabled, depth])

  return (
    <div
      data-testid="utility-icon"
      className="ruio-exclude"
      style={{
        position: 'fixed',
        bottom: '20px',
        right: '20px',
        zIndex: 9999,
        backgroundColor: bordersEnabled ? 'lightgreen' : 'lightcoral',
        padding: '10px',
        cursor: 'pointer',
      }}
      onClick={() => setBordersEnabled(!bordersEnabled)}
    >
      üõ†Ô∏è
    </div>
  )
}

export default UtilityIcon
