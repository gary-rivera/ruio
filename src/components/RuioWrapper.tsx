import React, { ReactNode } from 'react'
import { RuioContextProvider } from '@root/context/RuioContextProvider'
import ControlPanel from './ControlPanel'

// Single point of entry to ease the process of handling context throughout the module
const RuioWrapper: React.FC<{ children: ReactNode }> = ({ children }) => {
  return (
    <RuioContextProvider>
      {/* <RuioLogo /> */}
      {/* <UtilityIcon /> */}
      <ControlPanel />
      {children}
    </RuioContextProvider>
  )
}

export default RuioWrapper
