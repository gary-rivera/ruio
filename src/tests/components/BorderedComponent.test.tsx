import React from 'react'
import { render } from '@testing-library/react'
import { BorderedComponent } from '@tests/components/BorderedComponent'

describe('BorderedComponent Snapshot Tests', () => {
  test('matches the snapshot when borders are applied', () => {
    const { asFragment } = render(<BorderedComponent depth={2} apply={true} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('matches the snapshot when borders are not applied', () => {
    const { asFragment } = render(<BorderedComponent depth={2} apply={false} />)
    expect(asFragment()).toMatchSnapshot()
  })

  test('matches the snapshot with depth 0', () => {
    const { asFragment } = render(<BorderedComponent depth={0} apply={true} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
