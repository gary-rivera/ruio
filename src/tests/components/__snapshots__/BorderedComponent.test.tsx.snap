// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderedComponent Snapshot Tests matches the snapshot when borders are applied 1`] = `
<DocumentFragment>
  <div
    style="outline: 2px solid red;"
  >
    Bordered Component
  </div>
</DocumentFragment>
`;

exports[`BorderedComponent Snapshot Tests matches the snapshot when borders are not applied 1`] = `
<DocumentFragment>
  <div>
    Bordered Component
  </div>
</DocumentFragment>
`;

exports[`BorderedComponent Snapshot Tests matches the snapshot with depth 0 1`] = `
<DocumentFragment>
  <div
    style="outline: 2px solid red;"
  >
    Bordered Component
  </div>
</DocumentFragment>
`;
